tests/Unit/Evolution/Systems/Cce/Test_NewmanPenrose.cpp

```diff
@@ -23,8 +36,315 @@ void pypp_test_volume_weyl() {
                                     DataVector{num_pts});
 }
 
+namespace {
+
+struct InverseCubicEvolutionGauge {
+  using boundary_tags =
+      tmpl::list<Tags::EvolutionGaugeBoundaryValue<Tags::BondiJ>,
+                 Tags::EvolutionGaugeBoundaryValue<Tags::Dr<Tags::BondiJ>>,
+                 Tags::EvolutionGaugeBoundaryValue<Tags::BondiR>>;
+
+  using mutate_tags = tmpl::list<Tags::BondiJ, Tags::CauchyCartesianCoords,
+                                 Tags::CauchyAngularCoords>;
+  using argument_tags =
+      tmpl::append<boundary_tags,
+                   tmpl::list<Tags::LMax, Tags::NumberOfRadialPoints>>;
+
+  InverseCubicEvolutionGauge() = default;
+
+  void operator()(
+      const gsl::not_null<Scalar<SpinWeighted<ComplexDataVector, 2>>*> j,
+      const gsl::not_null<tnsr::i<DataVector, 3>*> /*cartesian_x_of_x_tilde*/,
+      const gsl::not_null<
+          tnsr::i<DataVector, 2, ::Frame::Spherical<::Frame::Inertial>>*>
+      /*angular_cauchy_coordinates*/,
+      const Scalar<SpinWeighted<ComplexDataVector, 2>>& boundary_j,
+      const Scalar<SpinWeighted<ComplexDataVector, 2>>& boundary_dr_j,
+      const Scalar<SpinWeighted<ComplexDataVector, 0>>& r,
+      const size_t /*l_max*/, const size_t number_of_radial_points) const {
+    const DataVector one_minus_y_collocation =
+        1.0 - Spectral::collocation_points<Spectral::Basis::Legendre,
+                                           Spectral::Quadrature::GaussLobatto>(
+                  number_of_radial_points);
+    for (size_t i = 0; i < number_of_radial_points; i++) {
+      ComplexDataVector angular_view_j{
+          get(*j).data().data() + get(boundary_j).size() * i,
+          get(boundary_j).size()};
+      // auto is acceptable here as these two values are only used once in the
+      // below computation. `auto` causes an expression template to be
+      // generated, rather than allocating.
+      const auto one_minus_y_coefficient =
+          0.25 * (3.0 * get(boundary_j).data() +
+                  get(r).data() * get(boundary_dr_j).data());
+      const auto one_minus_y_cubed_coefficient =
+          -0.0625 *
+          (get(boundary_j).data() + get(r).data() * get(boundary_dr_j).data());
+      angular_view_j =
+          one_minus_y_collocation[i] * one_minus_y_coefficient +
+          pow<3>(one_minus_y_collocation[i]) * one_minus_y_cubed_coefficient;
+    }
+  }
+};
+
+// This unit test is to validate the calculation of the Weyl scalar psi0 on the
+// worldtube. The structure is in parallel with Test_GaugeTransformBoundaryData
+// (most codes are copied from there). I construct a stationary Kerr spacetime
```

Comment:
"stationary Kerr spacetime" -> "stationary Kerr spacetime in nontrivial time-dependent oscillating coordinates"

Reply:
"I construct" -> "This constructions..."